java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -237,14 +239,25 @@ private ArrowBuf bufferWithoutReservation(final int size, BufferManager bufferMa
 
     final AllocationManager manager = new AllocationManager(this, size);
     final BufferLedger ledger = manager.associate(this); // +1 ref cnt (required)
-    final ArrowBuf buffer = ledger.newArrowBuf(0, size, bufferManager);
+    
+    int offset = calcOffset(ledger.getUnderlying());
+    final ArrowBuf buffer = ledger.newArrowBuf(offset, size, bufferManager);
```

Comment:
Yes, if we rely on netty's behavior. I thought about size, but I had some concerns about it. Let me check how netty will handle it.